### PR TITLE
Neighbourhood

### PR DESCRIPTION
**Pull request by smohsinali**

### CODE REVIEW

File: ConfigSpace/hyperparameters.py [hunk 1]
```diff
@@ -991,37 +991,43 @@ def get_num_neighbors(self, value: Union[int, float, str]) -> int:
         """
         returns the number of existing neighbors in the sequence
         """
-        if value == self.sequence[0] or value == self.sequence[-1]:
+        if value == self.sequence[0] and value == self.sequence[-1]:# check if there is only one value
+            return 0
+        elif value == self.sequence[0] or value == self.sequence[-1]:
```
Comment (mfeurer) [inline ConfigSpace/hyperparameters.py]: I think there is no check that the values in the sequence are unique. Could you please add that?

File: ConfigSpace/hyperparameters.py [hunk 2]
```diff
@@ -991,37 +991,43 @@ def get_num_neighbors(self, value: Union[int, float, str]) -> int:
...
-                    candidate2 = self.get_value(neighbor_idx2)
-                    if self.check_order(value, candidate2):
-                        neighbors.append(candidate2)
-            else:
-                if neighbor_idx1 < index and neighbor_idx1 >= seq[0]:
-                    neighbors.append(neighbor_idx1)
-                if neighbor_idx2 > index and neighbor_idx2 <= self._num_elements:
-                    neighbors.append(neighbor_idx2)
-
+            # todo: check if transform is needed
```
Comment (mfeurer) [inline ConfigSpace/hyperparameters.py]: I don't think it's necessary, but it is implemented for all other hyperparameters. Could you please remove this feature completely from all hyperparameters?

File: ConfigSpace/hyperparameters.py [hunk 3]
```diff
@@ -991,37 +991,43 @@ def get_num_neighbors(self, value: Union[int, float, str]) -> int:
...
+            #         candidate2 = self.get_value(neighbor_idx2)
+            #         if self.check_order(value, candidate2):
+            #             neighbors.append(candidate2)
+            # else:
+            if neighbor_idx1 < index and neighbor_idx1 >= seq[0]:
+                neighbors.append(neighbor_idx1)
+            if neighbor_idx2 > index and neighbor_idx2 < self._num_elements:
+                neighbors.append(neighbor_idx2)
+
+        # todo: is there a scenario where empty neighbors array is returned or there should be a check for it?
```
Comment (mfeurer) [inline ConfigSpace/hyperparameters.py]: No need to check this, if there are no neighbors, you can return an empty list.

File: test/test_hyperparameters.py [hunk 4]
```diff
@@ -634,27 +634,31 @@ def test_ordinal_check_order(self):
...
+        self.assertEqual(tuple(f1.get_seq_order()), tuple([0,1,2,3]))
     
     def test_ordinal_get_neighbors(self):
         f1 = OrdinalHyperparameter("temp", 
                                    ["freezing", "cold", "warm", "hot"])
-        self.assertEqual(f1.get_neighbors("freezing", rs=None), [2])
-        self.assertEqual(f1.get_neighbors("cold", transform=True, rs=None), ["freezing", "warm"])
-        self.assertEqual(f1.get_neighbors("hot", rs=None), [3])
-        self.assertEqual(f1.get_neighbors("hot", transform =True, rs=None), ["warm"])
+        # self.assertEqual(f1.get_neighbors("freezing", rs=None), [2])
```
Comment (mfeurer) [inline test/test_hyperparameters.py]: Please remove the commented code.

File: ConfigSpace/hyperparameters.py (hunk 2 above)
Comment (smohsinali) [inline ConfigSpace/hyperparameters.py]: done.

File: ConfigSpace/hyperparameters.py (hunk 3 above)
Comment (smohsinali) [inline ConfigSpace/hyperparameters.py]: done

File: test/test_hyperparameters.py (hunk 4 above)
Comment (smohsinali) [inline test/test_hyperparameters.py]: done

File: ConfigSpace/hyperparameters.py (hunk 1 above)
Comment (smohsinali) [inline ConfigSpace/hyperparameters.py]: need to discuss bit more

Comment (smohsinali) [inline ConfigSpace/hyperparameters.py]: done

File: test/test_hyperparameters.py (hunk 4 above)
Comment (smohsinali) [inline test/test_hyperparameters.py]: put back transforms because they are needed in get_random_neighbor function

**Pull request merged**